src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[INITIAL_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        growArray();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == elements.length) {
```

Comment:
let's add checking for index

Reply:
added
i had to add different checking6 not just using checkIndex() because index here can be the same value as size